internal/resolution/variablesources/olm_variable_source.go

```diff
@@ -0,0 +1,81 @@
+package variablesources
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+
+	operatorsv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	olmvariables "github.com/operator-framework/operator-controller/internal/resolution/variables"
+)
+
+type OLMVariableSource struct {
+	client        client.Client
+	catalogClient BundleProvider
+}
+
+func NewOLMVariableSource(cl client.Client, catalogClient BundleProvider) *OLMVariableSource {
+	return &OLMVariableSource{
+		client:        cl,
+		catalogClient: catalogClient,
+	}
+}
+
+func (o *OLMVariableSource) GetVariables(ctx context.Context) ([]deppy.Variable, error) {
+	operatorList := operatorsv1alpha1.OperatorList{}
+	if err := o.client.List(ctx, &operatorList); err != nil {
+		return nil, err
+	}
+
+	bundleDeploymentList := rukpakv1alpha1.BundleDeploymentList{}
+	if err := o.client.List(ctx, &bundleDeploymentList); err != nil {
+		return nil, err
+	}
+
+	allBundles, err := o.catalogClient.Bundles(ctx)
```

Comment:
would it make sense to return some kind of wrapper type around the list of bundles (something like `snapshot`...?) and attach the filtering methods to that, rather than relying on utility functions to do it?

Reply:
I also would prefer to keep things simple.